fondant/component_spec.py

```diff
@@ -18,12 +18,12 @@
 
 # TODO: remove after upgrading to kfpv2
 kubeflow2python_type = {
-    "String": str,
-    "Integer": int,
-    "Float": float,
-    "Boolean": ast.literal_eval,
-    "JsonObject": json.loads,
-    "JsonArray": json.loads,
+    "String": lambda value: str(value) if value != "None" else None,
+    "Integer": lambda value: int(value) if value != "None" else None,
+    "Float": lambda value: float(value) if value != "None" else None,
+    "Boolean": lambda value: ast.literal_eval(value) if value != "None" else None,
+    "JsonObject": lambda value: json.loads(value) if value != "None" else None,
+    "JsonArray": lambda value: json.loads(value) if value != "None" else None,
```

Comment:
Maybe it makes sense converting this into a function.

```suggestion
kubeflow_to_python_type_dict = {
    "String": str,
    "Integer": int,
    "Float": float,
    "Boolean": ast.literal_eval,
    "JsonObject": json.loads,
    "JsonArray": json.loads,
}

def kubeflow_to_python_type(type_: str) -> t.Any:
    map_fn = kubeflow_to_python_type_dict[type_]
    return lambda value: map_fn(value) if value != "None" else None
```

Reply:
or use the `match case` syntax: https://peps.python.org/pep-0622/

Reply:
Thanks for the feedback :) I think it's ready to go